presto-testing/src/main/java/io/prestosql/testing/datatype/DataType.java

```diff
@@ -202,6 +220,11 @@ public static String binaryLiteral(byte[] value)
         return new DataType<>(insertType, prestoResultType, Object::toString, Function.identity());
     }
 
+    public static <T> DataType<T> dataType(String insertType, Type prestoResultType, Function<T, String> toLiteral)
```

Comment:
Adding this overload may let you simplify call sites to the existing, 4-arg overload.

Reply:
for a followup